tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -495,35 +474,32 @@ def test_ad_integration(
     instance,
     os,
     pcluster_config_reader,
-    clusters_factory,
     directory_type,
     test_datadir,
     s3_bucket_factory,
     directory_factory,
     request,
     store_secret_in_secret_manager,
+    clusters_factory,
 ):
     """Verify AD integration works as expected."""
     compute_instance_type_info = {"name": "c5.xlarge", "num_cores": 4}
     config_params = {"compute_instance_type": compute_instance_type_info.get("name")}
-    ad_admin_password = "".join(random.choices(string.ascii_letters + string.digits, k=60))
-    ad_user_password = "".join(random.choices(string.ascii_letters + string.digits, k=60))
-    password_secret_arn = store_secret_in_secret_manager(ad_admin_password)
-    if directory_type:
-        bucket_name = s3_bucket_factory()
-        directory_stack_name, nlb_stack_name = directory_factory(
-            request.config.getoption("directory_stack_name"),
-            request.config.getoption("ldaps_nlb_stack_name"),
-            directory_type,
-            bucket_name,
-            str(test_datadir),
-            region,
-            ad_admin_password=ad_admin_password,
-            ad_user_password=ad_user_password,
-        )
-        config_params.update(
-            get_ad_config_param_vals(directory_stack_name, nlb_stack_name, bucket_name, password_secret_arn)
-        )
+    bucket_name = s3_bucket_factory()
+    directory_stack_name, nlb_stack_name = directory_factory(
+        request.config.getoption("directory_stack_name"),
+        request.config.getoption("ldaps_nlb_stack_name"),
+        directory_type,
+        bucket_name,
+        str(test_datadir),
+        region,
+    )
+    directory_stack_outputs = get_infra_stack_outputs(directory_stack_name)
+    ad_user_password = directory_stack_outputs.get("UserPassword")
+    password_secret_arn = store_secret_in_secret_manager(directory_stack_outputs.get("AdminPassword"))
+    config_params.update(
+        get_ad_config_param_vals(directory_stack_name, nlb_stack_name, bucket_name, password_secret_arn)
```

Comment:
Minor: Why don't we put `ad_user_password`, `password_secret_arn` as the return value of the directory_factory stack?